### PR TITLE
feat: #115 Add Checkbox component

### PR DESCRIPTION
**Pull request by Shustovsky**

### CODE REVIEW

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   * The name of the checkbox.
+   */
+  name?: string;
+
+  /**
+   * The value of the checkbox.
+   * @default "on"
+   */
+  value?: string;
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:36]: Can you pls explain what doing this prop? Why default value is on? How it depends on checkbox?

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * The checked state of the checkbox when it is initially rendered.
+   * @default false
+   */
+  defaultChecked?: boolean;
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:42]: If this prop only for default state of checkbox then why we need it. If it prop will be change by some code then rename required

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * When true, prevents the user from interacting with the checkbox.
+   * @default false
+   */
+  disabled?: boolean;
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:48]: I understand that checkbox can be disabled. ButI think that  in our app  it's almost impossible. Maybe we no need this prop?

Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:48]: I understand that checkbox can be disabled. ButI think that  in our app  it's almost impossible. Maybe we no need this prop?

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * The required property a checkbox must be checked before submitting a form.
+   * @default false
+   */
+  required?: boolean;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:53]: Pls give an example where we will need checkbox required?

Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:53]: Pls give an example where we will need checkbox required?

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:54]: You forgot add onChange prop

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 4 above)
Comment (Howl404) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:53]: Probably an agreement with some privacy terms and etc

File: mw-webapp/src/styles/_variables.scss [hunk 6]
```diff
@@ -40,3 +43,10 @@ $sidebar-button-offset: 5px;
 // Size
 $width-primary: 100%;
 $height-full: 100%;
+
+// Checkbox
```
Comment (Howl404) [inline mw-webapp/src/styles/_variables.scss:47]: Can we group Size, Color and Font Size styles together on the bottom of the file? 
Now I see that they are starting to get lost among the component styles

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/styles/_variables.scss (hunk 6 above)
Comment (Howl404) [reply mw-webapp/src/styles/_variables.scss:47]: Or maybe vice versa to the very top

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 1 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:36]: this prop is not needed in the project at this stage, I removed it

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 2 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:42]: this prop is not needed in the project at this stage, I removed it

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 3 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:48]: this prop is not needed in the project at this stage, I removed it

Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:48]: this prop is not needed in the project at this stage, I removed it

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 4 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:53]: this prop is not needed in the project at this stage, I removed it

Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:53]: this prop is not needed in the project at this stage, I removed it

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 5 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:54]: yeah, of course)  add it)

File: mw-webapp/src/styles/_variables.scss (hunk 6 above)
Comment (Shustovsky) [reply mw-webapp/src/styles/_variables.scss:47]: I'll do it later in the task [#41](https://github.com/tritonJS826/masters-way/issues/41 )

**Review by Howl404 — APPROVED** [review]

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 5 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx:23]: Are you sure that the callback should not accept any parameters? 
If not = pls explain why, if yes - pls explain why

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  /**
+   * Indicates whether the checkbox is checked.
+   */
+  checked: boolean;
+
+  /**
+   * A callback function to be called when the checkbox's state changes.
+   */
+  onChange: (value?: boolean) => void;
```
Comment (tritonJS826) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:22]: as I remember we discussed signature of onChange method and agreed on required value parameter, fix please

File: mw-webapp/src/component/checkbox/Сheckbox.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Checkbox component
+ */
+export const Checkbox = (props: CheckboxProps) => {
+  return (
+    <input
+      className={clsx(styles.checkbox, props.className)}
+      type="checkbox"
+      checked={props.checked}
+      onChange={(event) => props.onChange(!!event.target.value)}
```
Comment (tritonJS826) [inline mw-webapp/src/component/checkbox/Сheckbox.tsx:34]: what about:
```suggestion
      onChange={props.onChange}
```

**Review by tritonJS826 — CHANGES_REQUESTED** [review]
I will sent you a video

File: mw-webapp/src/component/checkbox/Сheckbox.tsx (hunk 7 above)
Comment (Shustovsky) [reply mw-webapp/src/component/checkbox/Сheckbox.tsx]: i fix it

**Review by Ekaterina1994 — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]
Yeah, nice)

**Pull request merged**